dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04200CreateDefaultVanityURL.java

```diff
@@ -0,0 +1,235 @@
+package com.dotmarketing.startup.runonce;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+
+/**
+ * This upgrade task will insert into the tables Inode and Structure the default Vanity URL Content Type.
+ * This is needed to be able to migrate the old Vanity URL's (on the virtual_link table) to the new Vanity URL that are content.
+ * 
+ * @author erickgonzalez
+ * @version 4.2.0
+ * @since Jun 6, 2017
+ *
+ */
+public class Task04200CreateDefaultVanityURL extends AbstractJDBCStartupTask {
+
+	// TODO: If the Content Type is changed we need to check the values here are the same that the ones on the Starter
+	private final String DEFAULT_VANITY_URL_STRUCTURE_INODE = "8e850645-bb92-4fda-a765-e67063a59be0";
+	private final String DEFAULT_VANITY_URL_STRUCTURE_VARNAME = "Vanityurlasset";
+	private final String DEFAULT_VANITY_URL_STRUCTURE_NAME = "Vanity URL Asset";
+	private final String DEFAULT_VANITY_URL_STRUCTURE_DESCRIPTION = "Vanity URL as Content";
+	private final String[] DEFAULT_VANITY_URL_FIELDS_INODES = { 
+			"4e64a309-8c5e-48cf-b4a9-e724a5e09575", // Order
+			"3b7f25bb-c5be-48bb-b00b-5b1b754e550c", // Action
+			"de4fea7f-4d8f-48eb-8a63-20772dced99a", // Foward To
+			"7e438b93-b631-4812-9c9c-331b03e6b1cd", // Uri
+			"49f3803a-b2b0-4e03-bb2e-d1bb2a1c135e", // Site
+			"9ae3e58a-75c9-4e4a-90f3-6f0d5f32a6f0"  // Title
+	};
+
+	private final String INSERT_INODE_QUERY = "insert into inode (inode, owner, idate, type) values (?, ?, ?, ?)";
+
+	private final String INSERT_FIELD_QUERY = "insert into field "
+			+ "(inode,structure_inode,field_name,field_type,field_relation_type,field_contentlet,"
+			+ "required,indexed,listed,velocity_var_name,sort_order,field_values,regex_check,"
+			+ "hint,default_value,fixed,read_only,searchable,unique_,mod_date) "
+			+ "VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)";
+
+	private final String INSERT_STRUCTURE_QUERY = "insert into structure (inode, name, description, structuretype, system, fixed, velocity_var_name, host, "
+			+ "folder, mod_date, default_structure) " + "values ( ?, ?, ?, ?, ?, ?, ?, ?, ? ,?,?)";
+
+	@Override
+	public void executeUpgrade() throws DotDataException, DotRuntimeException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)